.travis.yml

```diff
@@ -0,0 +1,67 @@
+language: c
+compiler:
+  - gcc
+  - clang
+
+arch: amd64
+os: linux
+dist: bionic
+
+env:
+  global:
+    - JANUS_COMMON_OPTS="--prefix=/opt/janus --disable-docs --enable-post-processing --enable-plugin-lua --enable-plugin-duktape --enable-json-logger"
+  jobs:
+    - JANUS_CONFIG_OPTS="" LIBCURL=YES
+    - JANUS_CONFIG_OPTS="--disable-data-channels" LIBCURL=YES
+    - JANUS_CONFIG_OPTS="--disable-turn-rest-api --disable-sample-event-handler" LIBCURL=NO
+
+git:
+  quiet: true
+
+addons:
+  apt:
+    update: true
+    packages:
+      - autoconf
+      - cmake
+      - gengetopt
+      - gtk-doc-tools
+      - libavcodec-dev
+      - libavformat-dev
+      - libavutil-dev
+      - libcollection-dev
+      - libconfig-dev
+      - libevent-dev
+      - libglib2.0-dev
+      - libgnutls28-dev
+      - libini-config-dev
+      - liblua5.3-dev
+      - libjansson-dev
+      - libmicrohttpd-dev
+      - libmount-dev
+      - libnanomsg-dev
+      - libogg-dev
+      - libopus-dev
+      - libsofia-sip-ua-dev
+      - libssl-dev
+      - libvorbis-dev
+      - openssl
+
+before_script:
+  - if [ $LIBCURL = YES ]; then sudo apt-get -y install libcurl4-openssl-dev; fi
+  - git clone -b master https://gitlab.freedesktop.org/libnice/libnice.git libnice
+  - pushd libnice && ./autogen.sh && ./configure --prefix=/usr --disable-gtk-doc --disable-gtk-doc-html --disable-gupnp && make -j$(nproc) && sudo make install && popd
+  - git clone -b 2_2_x_throttle https://github.com/cisco/libsrtp.git libsrtp
+  - pushd libsrtp && ./configure --prefix=/usr --enable-openssl && make -j$(nproc) shared_library && sudo make install && popd
+  - git clone -b master https://github.com/sctplab/usrsctp usrsctp
+  - pushd usrsctp && ./bootstrap && ./configure --prefix=/usr --disable-static --disable-debug --disable-programs && make -j$(nproc) && sudo make install && popd
+  - git clone -b v3.2-stable https://github.com/warmcat/libwebsockets.git libwebsockets
+  - pushd libwebsockets && mkdir -p build && pushd build && cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DLWS_WITH_STATIC=OFF -DLWS_WITHOUT_CLIENT=ON -DLWS_WITHOUT_TESTAPPS=ON -DLWS_WITHOUT_TEST_SERVER=ON -DLWS_WITH_HTTP2=OFF .. && make -j$(nproc) && sudo make install && popd && popd
+  - git clone -b master https://github.com/alanxz/rabbitmq-c rabbitmq-c
+  - pushd rabbitmq-c && git submodule update --init && mkdir build && pushd build && cmake -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_EXAMPLES=OFF -DBUILD_TESTS=OFF -DBUILD_TOOLS=OFF .. && sudo cmake --build . --target install -- -j$(nproc) && popd && popd
+  - git clone -b master https://github.com/eclipse/paho.mqtt.c.git paho.mqtt.c
+  - pushd paho.mqtt.c && cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DPAHO_WITH_SSL=TRUE -DPAHO_BUILD_SAMPLES=FALSE -DPAHO_BUILD_DOCUMENTATION=FALSE && make -j$(nproc) && sudo make install && popd
```

Comment:
You may want to replace some of these clone commands with e.g., `curl -fSL https://github.com/warmcat/libwebsockets/archive/v${LIBWEBSOCKETS_VERSION}.tar.gz`.

 Also sticking to specific versions may be easier to debug (but github does let you grab tarballs of `master` or arbitrary  .commits too).

Otherwise, if you want to keep doing `git clone` I'd suggest adding `--depth 1`.

Reply:
+1 for `--depth 1`. I don't see any advantage in getting rid of `git clone` in favor of getting a tarball via http.

Reply:
Yeah it'd only be in a situation where you didn't want the extra dependency of git or were doing http caching (via a proxy) and so needed to download over plain http.